### ISSUE TITLE
x/vuln: govulncheck v1.0.2 json output content is different to regular output (10s more false positives)

### ISSUE THREAD

**Issue by inbarassaf** [golang/go#65132]
When running govulncheck v1.0.2 on any project (event a simple hello world, see example attached), the output when running with json flag is much longer (contains many more vulnerabilities- false positives) than when running without the json flag.

Example: I've created a simple hello world project with the following mod file:
`module simple_vuln_check/hello

go 1.21.5
`

and the following main file:
`package main

import "fmt"

func main() {
	fmt.Println("Hello, World!")
}
`

And run the following:
`govulncheck  . ` in the directory the project, to get this output (as expected, no vulnerabilities):
`Scanning your code and 42 packages across 0 dependent modules for known vulnerabilities...

No vulnerabilities found.

Share feedback at https://go.dev/s/govulncheck-feedback.
`

But when running with the json flag:
`govulncheck -json .`
I get the output in the file attached (containing 93 vulnerabilities)
[simple_vuln_json.txt](https://github.com/golang/go/files/13962740/simple_vuln_json.txt)

This seems like a serious regression in behaviour- this does not happen in 1.0.1

**Comment by zpavlinovic** [comment]
It is not a regression per se, yet an enhancement of the streaming capabilities of govulncheck `-json`.

govulncheck json tries to communicate vulnerability information the moment it discovers it, which is useful for large projects where govulncheck might take time.

It will first communicate all vulnerabilities (OSVs) known for the modules your project requires.

It will then see which of these actually apply to your modules given the module version you depend on. These will be communicated as findings with only module information known.

It will then see which of these vulnerabilities are imported by your project; these will be findings with package and module information set.

Finally, it will try to see which of these vulnerabilities are actually called; these will be findings with symbol, package, and module information set.

Skimming through your file, I can only see osvs being reported.

**Comment by zpavlinovic** [comment]
Yes, you should see the same osvs for the program at hand. Binary mode will in general produce less findings though.

We'll add more documentation on this.
